src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs

```diff
@@ -61,10 +69,16 @@ internal FormulaValue Process()
 
                 using MemoryStream memoryStream = new MemoryStream();
                 using Utf8JsonWriter writer = new Utf8JsonWriter(memoryStream, new JsonWriterOptions() { Indented = flags.IndentFour, Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping });
-                Utf8JsonWriterVisitor jsonWriterVisitor = new Utf8JsonWriterVisitor(writer, _timeZoneInfo, flattenValueTables: flags.FlattenValueTables);
+                Utf8JsonWriterVisitor jsonWriterVisitor = new Utf8JsonWriterVisitor(writer, _timeZoneInfo, flattenValueTables: flags.FlattenValueTables, canceller);
 
-                _arguments[0].Visit(jsonWriterVisitor);
-                writer.Flush();
+                try
+                {
+                    _arguments[0].Visit(jsonWriterVisitor);
+                    writer.Flush();
+                }
+                catch (InvalidOperationException)
```

Comment:
>InvalidOperationException [](http://example.com/codeflow?start=23&length=25)

Can you add a comment saying why this exception is being ignored? Or we can move the error logic (lines 83-91) into the catch block, something like

    catch (InvalidOperationException) {
        if (writer.ErrorValues.Any()) {
            // logic to return error
        }

        // Unexpected error, rethrow
        throw;
    }